mantidproject/mantid
# Discus Multiple Scattering Usability Enhancements

**Description of work.**

This is a slightly miscellaneous set of improvement\bug fixes to the inelastic version of the Discus multiple scattering correction. These are from feedback and other issues I've noticed since this sizeable change went in a few weeks ago as part of PR https://github.com/mantidproject/mantid/pull/33609:

- fix validation on ScatteringCrossSection parameter
- remove implicit assumption that w bins in S(Q,w) all have to be same size
- include results workspace group name in the member workspace names. If user runs any of the output workspaces through any further Mantid algorithms it's easier to see which workspace group they come from
- add some extra output workspaces (an extra "summed_1_n" workspace and Integrated workspaces where weights have been integrated across the x axis)
- Fix bug where vmu variable wasn't being updated after each scatter. Only affects inelastic calculations - effect on results is v small
- Remove a "2" from the equations in the documentation page. This shouldn't there and wasn't in the code
- Update Absorption\Multiple Scattering concepts page - this still said the algorithm was elastic only

**To test:**

Run the script on https://github.com/mantidproject/mantid/pull/33609
Check it completes successfully and also that the workspaces inside the results workspace group have names that all begin "MuscatResults..."

*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
